CI improvements

bamarni opened the pull request:
- bump python to 3.6
- Run Travis CI builds on MacOS too
- Add flake8 / pylint and fix code accordingly
- Add 'make lint' target

bamarni committed:
Run Travis CI builds on MacOS too

bamarni committed:
Add flake8 and fix code accordingly

bamarni committed:
Small README updates

adamtheturtle commented on .travis.yml:
```diff
@@ -1,8 +1,22 @@
+
+matrix:
+  include:
+    - os: linux
```
Keep in mind that the current CLI supports Windows which is not supported by Travis CI.

bamarni replied:
That's right, I will take a look at https://www.appveyor.com in the next steps.

adamtheturtle commented on .travis.yml:
```diff
@@ -1,8 +1,22 @@
+    - os: osx
+      # workaround for https://github.com/travis-ci/travis-ci/issues/2312
+      language: generic
+      before_install:
```
Does this mean that macOS builds may have a different version of Python to Linux builds, and importantly, different to what the documentation says is required?

bamarni replied:
Good point, it seems to be 3.6 from the build console logs. I'll look for a way to stick to 3.5 there or to bump requirements (https://github.com/bamarni/dcos_cli/issues/2).

adamtheturtle commented on Makefile:
```diff
@@ -2,6 +2,11 @@ deps:
-test:
+lint:
+	flake8 --verbose dcos_cli/ tests/
+	pylint --disable=missing-docstring,cyclic-import dcos_cli/ tests/
```
`pylint` configuration inevitably becomes long. I recommend using a pylint configuration file, but that can wait 😄

adamtheturtle commented on dcos_cli/cli/cluster/__init__.py:
```diff
@@ -1,14 +1,14 @@
-import dcos_cli.cli.cluster.rename # noqa: E402
-import dcos_cli.cli.cluster.setup # noqa: E402
+# pylint: disable=wrong-import-position
+import dcos_cli.cli.cluster.attach # noqa
```
Why remove the specifics of the `noqa`? Using specific codes allows `flake8` to report issues that you didn't mean to ignore.

I don't understand why these are imported late. Please add a comment if there is a good reason, or move them to the top of the file.

bamarni replied:
I was missing the noqa F401. As for these imports it's the same reason than here (https://github.com/mesosphere/dcos-security-cli/blob/master/dscli/__init__.py#L99).

I will check to see if there is another way to do this, that'd also allow to get rid of linter exceptions for them.

bamarni replied:
I've switched to another way, cf. last commit.

adamtheturtle replied:
🙏

bamarni committed:
Switch to click.add_command()

This allows not to put exceptions for various pylint / flake8 errors
complaining about unused / cyclic imports

bamarni committed:
Rely on Python 3.6 (closes #2)

bamarni commented:
@adamtheturtle : I'll merge this on once the build is green, as I think I've addressed all your comments, thanks for the review.
